### ISSUE TITLE
Mixed content in DITA codeblocks wraps at print width

### ISSUE THREAD

**Issue by infotexture** [prettier/plugin-xml#100]
The DITA [`<codeblock>`](http://docs.oasis-open.org/dita/dita/v1.3/errata02/os/complete/part3-all-inclusive/langRef/technicalContent/codeblock.html#codeblock) element is designed for code samples where line breaks are significant.

## Works as expected if codeblocks don't contain other XML elements

The Prettier XML plugin seems to do the right thing here most of the time, wrapping _attributes_ of the `<codeblock>` element to a new line to meet the print-width requirements, but the element _content_ is left alone, so the result is functionally equivalent when running with the following options:

```bash
prettier
  --write "**/*.dita"
  --parser=xml
  --print-width 120
  --xml-self-closing-space false
  --xml-whitespace-sensitivity ignore
```

### Before

```xml
<codeblock outputclass="language-xml normalize-space show-line-numbers show-whitespace">&lt;plugin id="com.example.WithSupportInfo">
```

### After

```xml
<codeblock
  outputclass="language-xml normalize-space show-line-numbers show-whitespace"
>&lt;plugin id="com.example.WithSupportInfo">
```

## Fails to preserve lines if codeblock contains other elements

Given a codeblock with inline elements, lines are wrapped at the print width instead of preserving the original lines.

### Before

```xml
<codeblock outputclass="language-bash"><cmdname>dita uninstall</cmdname> <varname>&lt;plug-in-id&gt;</varname></codeblock>
```

The results differ depending on the `--xml-whitespace-sensitivity` setting, but neither manages to preserve the line as authored:

### After --xml-whitespace-sensitivity ignore

```xml
<codeblock outputclass="language-bash">
  <cmdname>dita uninstall</cmdname>
  <varname>&lt;plug-in-id&gt;</varname>
</codeblock>
```

### After --xml-whitespace-sensitivity strict

```xml
<codeblock outputclass="language-bash"><cmdname>dita uninstall</cmdname> <varname
  >&lt;plug-in-id&gt;</varname></codeblock>
```

**Comment by infotexture** [comment]
@kddeisz Does the XML plugin honor any magic comment syntax that can be used to ignore a range of lines like the [Range Ignore](https://prettier.io/docs/en/ignore.html#range-ignore) offered for top-level Markdown content?

Something like that would provide a workaround for cases like this:

```xml
<!-- prettier-ignore-start -->
<codeblock outputclass="language-bash"><cmdname>dita uninstall</cmdname> <varname>&lt;plug-in-id&gt;</varname></codeblock>
<!-- prettier-ignore-end -->
```

**Comment by kddeisz** [comment]
Just merged ignore ranges here: https://github.com/prettier/plugin-xml/pull/102. It's out in v0.9.0. So you could use that.

Are you saying even in `<varname\n>` the newline within there is significant? I guess in my head it couldn't be significant if it was within a start tag like that.

**Comment by infotexture** [comment]
> Just merged ignore ranges here: #102. It's out in v0.9.0. So you could use that.

🎉 Excellent! Thanks, that's a pragmatic workaround for now. 👍 
 
> Are you saying even in `<varname\n>` the newline within there is significant? 
> I guess in my head it couldn't be significant if it was within a start tag like that.

Normally with DITA codeblocks, **all** line breaks in source are passed to output 1:1. 

The `<cmdname>` element applies **bold** formatting and `<varname>` _italics_, but the lines break as specified in source.

The original source _(where the entire codeblock is on one line in the source file)_ is output correctly as:

> **dita uninstall** _`<plug-in-id>`_

The modified source from `--xml-whitespace-sensitivity ignore` mode is output on separate lines with additional indentation:

>         **dita uninstall**
>         _`<plug-in-id>`_

When I saw `--xml-whitespace-sensitivity strict` mode also breaking lines within the codeblock content, I assumed this would also affect output.

— But on closer inspection, it looks like line breaks are indeed only inserted in places that are **not significant** when generating output, so `strict` mode is probably working as designed. 

**TL;DR:** Closing as user error.
